src/Translator/src/Intl/ErrorKind.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Symfony\UX\Translator\Intl;
+
+/**
+ * Adapted from https://github.com/formatjs/formatjs/blob/590f1f81b26934c6dc7a55fff938df5436c6f158/packages/icu-messageformat-parser/error.ts#L9-L77
+ */
+final class ErrorKind
```

Comment:
Did you adapt these classes? Or were they in the bazinga package?

Reply:
Classes in namespace `Symfony\UX\Translator\Intl` have been adapted from the https://github.com/formatjs/formatjs/tree/590f1f81b26934c6dc7a55fff938df5436c6f158/packages/icu-messageformat-parser JS package to PHP, as documented in the PHPDoc, but maybe this is not enough.

Reply:
Is that so that you can dump the translations in JS?

Reply:
Yes and no, without this I can still dump translations in JS, but Intl ICU messages won't have definition types for their parameters.

To support parameters extraction (and their possible values for `select`/`plural` control blocks), I had to migrate the JS package `icu-messageformat-parser` to PHP, since there is no existing library doing this (either `Intl` from PHP or `symfony/intl`).


<img width="891" alt="image" src="https://user-images.githubusercontent.com/2103975/212642486-574ed272-2fc7-404a-a0da-0b26663117c3.png">

Given this complex ICU message, we were able to extract the following parameters:
- `gender_of_host`, a string, which can takes `female`, `male` or anything else (because `other` is required) 
- `num_guest`, a number
- `host`, a string
- `guest`, a string